File: odroid/Bert/src/chuckcoughlin/bert/robot/Controller.java
/**
 * Copyright 2018. Charles Coughlin. All Rights Reserved.
 * MIT License.
 */
package chuckcoughlin.bert.robot;

import java.util.List;


/**
 *  This interface describes a controller that holds a  list of motors.
 */
public interface Controller  {
	List<Motor> getMotors();
}

File: Bert/common/src/main/kotlin/chuckcoughlin/bert/common/controller/Controller.kt
/**
 * Copyright 2022-2023. Charles Coughlin. All Rights Reserved.
 * MIT License.
 */
package chuckcoughlin.bert.common.controller

/**
 * A common interface both for controllers that communicate with
 * the dispatcher (through channels) and also with external entities.
 * Channel naming is from the point of view of the current class.
 */
interface Controller {
    val controllerName: String
    val controllerType: ControllerType

    suspend fun execute()
    // Closes all resources including channels
    suspend fun shutdown()
}